src/webgpu/api/validation/capability_checks/limits/maxStorageBuffersPerShaderStage.spec.ts

```diff
@@ -0,0 +1,142 @@
+import { range, reorder, kOrderKeys } from '../../../../../common/util/util.js';
+import { GPUConst } from '../../../../constants.js';
+
+import {
+  kLimitBaseParams,
+  makeLimitTestGroup,
+  kBindGroupTests,
+  kBindingCombinations,
+  getPipelineTypeForBindingCombination,
+  getPipelineAsyncTypeForBindingCombination,
+  getPerStageWGSLForBindingCombination,
+} from './limit_utils.js';
+
+const limit = 'maxStorageBuffersPerShaderStage';
+export const { g, description } = makeLimitTestGroup(limit);
+
+g.test('createBindGroupLayout,at_over')
+  .desc(
+    `
+  Test using at and over ${limit} limit in createBindGroupLayout
+  
+  Note: We also test order to make sure the implementation isn't just looking
+  at just the last entry.
+  `
+  )
+  .params(
+    kLimitBaseParams
+      .combine('visibility', [
+        GPUConst.ShaderStage.FRAGMENT,
+        GPUConst.ShaderStage.COMPUTE,
+        GPUConst.ShaderStage.FRAGMENT | GPUConst.ShaderStage.COMPUTE,
+      ])
+      .combine('type', ['storage', 'read-only-storage'] as GPUBufferBindingType[])
```

Comment:
We could test readonlystorage for vertex as well. It should be allowed.

Reply:
added